src/Rebing/GraphQL/Support/SelectFields.php

```diff
@@ -37,8 +38,25 @@ public function __construct(ResolveInfo $info, $parentType, array $args)
         if( ! is_null($info->fieldNodes[0]->selectionSet))
         {
             self::$args = $args;
+            $requestedFields = $info->getFieldSelection(5);
+            $parentTypeName = get_object_vars($parentType)['config']['name'];
+
+            // If use pagination,
+            // $parentType & $requestedFields should use QueryType
+            // instead of paginationType
+            $paginationType = config('graphql.pagination_type', PaginationType::class);
```

Comment:
This whole section looks overly complicated. For example, why do you use `get_object_vars($parentType)...` instead of just `$parentType->config ...`? It also seems like there are too many assumptions that some array keys will always exist

Reply:
updated, also change the way to get QueryType